src/stemmabench/algorithms/stemma_NJ.py

```diff
@@ -0,0 +1,161 @@
+from typing import Callable, Dict, Union, Tuple
+import numpy as np
+from stemmabench.algorithms.stemma_algorithm import StemmaAlgo
+from stemmabench.algorithms.manuscript_in_tree_base import ManuscriptInTreeBase
+from stemmabench.algorithms.manuscript_in_tree_empty import ManuscriptInTreeEmpty
+from stemmabench.algorithms.utils import Utils
+
+
+class StemmaNJ(StemmaAlgo):
+    """Class that constructs a stemma using the Neighbor-Joining algorithm.
+
+    ### Attributes:
+        - folder_path (str): The path to the folder containing all the texts.
+        - manuscripts (dict): The dictionay of all the texts with text labels as keys and texts as values.
+        - distance (Callable): The function to be used as a distance metric.
+        - _dist_matrix (numpy.ndarray): The distance matrix
+    """
+
+    def __init__(self,
+                 distance: Union[Callable, None] = None) -> None:
+        """
+        Constructor for the StemmaNJ class.
+
+        ### Args:
+            - distance (Callable, Optional): A function that takes 2 strings as parameters and returns a numeric value which is
+            the distance between the 2 strings.
+        """
+        super().__init__()
```

Comment:
même pas besoin :-)

Reply:
Removed

Reply:
Does not seem to work properly if it is removed